api/src/DataPersister/Util/AbstractDataPersister.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace App\DataPersister\Util;
+
+use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;
+use App\Entity\BaseEntity;
+use InvalidArgumentException;
+
+abstract class AbstractDataPersister implements ContextAwareDataPersisterInterface {
+    /**
+     * @param CustomActionListener[]   $customActionListeners
+     * @param PropertyChangeListener[] $propertyChangeListeners
+     */
+    public function __construct(
+        private string $classname,
+        private DataPersisterObservable $dataPersisterObservable,
+        private array $customActionListeners = [],
+        private array $propertyChangeListeners = []
+    ) {
+        foreach ($this->customActionListeners as $listener) {
+            if (!$listener instanceof CustomActionListener) {
+                throw new InvalidArgumentException('customActionListeners must be of type '.CustomActionListener::class);
+            }
+        }
+        foreach ($this->propertyChangeListeners as $listener) {
+            if (!$listener instanceof PropertyChangeListener) {
+                throw new InvalidArgumentException('propertyChangeListeners must be of type '.PropertyChangeListener::class);
+            }
+        }
+    }
+
+    public function supports($data, array $context = []): bool {
+        return ($data instanceof $this->classname) && $this->dataPersisterObservable->supports($data, $context);
+    }
+
+    public function persist($data, array $context = []) {
+        $observable = $this->dataPersisterObservable
+            ->onBeforeCreate(fn ($data) => $this->beforeCreate($data))
```

Comment:
Auch hier allenfalls PHP callable Syntax möglich?

Reply:
du meinst [$this, 'beforeCreate'] ?
Willst du noch nicht extensiv lambda brauchen?

Hat bei mir direkt nicht funktioniert, hat bei mir aber auch schon sogar mit autocompletion in phpstorm funktioniert.
Geht das auch gut in VSCode?